tests/secp256k1_ecdsa.rs

```diff
@@ -59,3 +59,18 @@ fn test_secp256k1_sign_verify_random() {
     let pk = sk.public_key();
     run_secp256k1_sign_verify(sk, pk);
 }
+
+#[test]
+fn test_secp256k1_public_key_bytes() {
+    let mut sk_bytes = [0_u8; 32];
+    let mut pk_bytes = [0_u8; 33];
+    sk_bytes[0] += 1;
+    let sk = SecretKey::try_from_bytes(sk_bytes).unwrap();
+    let pk = sk.public_key();
+    pk_bytes.copy_from_slice(&pk.to_bytes());
+    assert_eq!(2, pk_bytes[0]);
+    pk_bytes[0] = 5;
+    // This is a SEC1 Compact form of the same public key. It is forbidden
+    let pk2 = PublicKey::try_from_bytes(&pk_bytes);
+    assert!(pk2.is_err());
```

Comment:
Actually checking the error would be even better, just to be sure another error is not taking precedence for some reason

Reply:
You mean like having separate error values for different error conditions?

Reply:
I mean, you should test that the error is the one you expect, because it could be something else if the test itself is malformed etc...

Reply:
The errors are not very meaningful, `try_from_bytes` returns `ConvertError`. Probably need to test for correct errors in other tests too.

Reply:
Yeah I guess it's contextual. Maybe it doesn't make sense for this one idk. Just saying that if we don't check the error, it could actually be another one due to an unrelated reason. But no worries.